pkg/oci/load_balancer_security_lists.go

```diff
@@ -531,8 +564,21 @@ func getLoadBalancerEgressRules(
 
 	egressRules := []core.EgressSecurityRule{}
 	for _, rule := range rules {
+		// Remove (do not re-add) any rule that represents the old case when mutating a single ranged port.
```

Comment:
This is the second part of the fix. I wanted to be on the safe side so made it very restrictive. It should only target the 'single port rule mutation' described.